src/app/components/image-viewer/ImageController.tsx

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable @typescript-eslint/no-shadow */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+
+import React, { useLayoutEffect, useState } from 'react';
+import { as } from 'folds';
+import * as css from './ImageViewer.css';
+import { Image } from './Image';
+import { Pan, usePan } from '../../hooks/usePan';
+import { useWheelZoom } from '../../hooks/useWheelZoom';
+
+export interface ImageControllerProps {
+  alt: string;
+  src: string;
+  zoom: number;
+  pan: Pan;
+
+  onChangeZoom: React.Dispatch<number>;
+  onChangePan: React.Dispatch<Pan>;
+}
+
+export const ImageController = as<'div', ImageControllerProps>(
+  ({ alt, src, zoom, pan, onChangeZoom, onChangePan, ...props }) => {
+    const [imageContainer, setImageContainer] = useState<HTMLDivElement | null>(null);
```

Comment:
using component reference in `useState` can lead to bugs, it should be replaced with `useRef` hook.